src/event/metric.rs

```diff
@@ -205,20 +225,139 @@ pub enum StatisticKind {
 }
 
 impl Metric {
+    pub fn new(
+        name: String,
+        namespace: Option<String>,
+        timestamp: Option<DateTime<Utc>>,
+        tags: Option<MetricTags>,
+        kind: MetricKind,
+        value: MetricValue,
+    ) -> Self {
+        Self {
+            series: MetricSeries {
+                name: MetricName { name, namespace },
+                tags,
+            },
+            data: MetricData {
+                timestamp,
+                kind,
+                value,
+            },
+        }
+    }
+
     /// Create a new Metric from this with all the data but marked as absolute.
     pub fn to_absolute(&self) -> Self {
         Self {
-            name: self.name.clone(),
-            namespace: self.namespace.clone(),
+            series: self.series.clone(),
+            data: self.data.to_absolute(),
+        }
+    }
+
+    /// Convert the metrics_runtime::Measurement value plus the name and
+    /// labels from a Key into our internal Metric format.
+    pub fn from_metric_kv(key: &metrics::Key, handle: &metrics_util::Handle) -> Self {
+        let value = match handle {
+            metrics_util::Handle::Counter(_) => MetricValue::Counter {
+                value: handle.read_counter() as f64,
+            },
+            metrics_util::Handle::Gauge(_) => MetricValue::Gauge {
+                value: handle.read_gauge() as f64,
+            },
+            metrics_util::Handle::Histogram(_) => {
+                let values = handle.read_histogram();
+                // Each sample in the source measurement has an
+                // effective sample rate of 1.
+                let samples = values
+                    .into_iter()
+                    .map(|i| Sample {
+                        value: i as f64,
+                        rate: 1,
+                    })
+                    .collect();
+                MetricValue::Distribution {
+                    samples,
+                    statistic: StatisticKind::Histogram,
+                }
+            }
+        };
+
+        let labels = key
+            .labels()
+            .map(|label| (String::from(label.key()), String::from(label.value())))
+            .collect::<MetricTags>();
+
+        Self::new(
+            key.name().to_string(),
+            Some("vector".to_string()),
+            Some(Utc::now()),
+            if labels.is_empty() {
+                None
+            } else {
+                Some(labels)
+            },
+            MetricKind::Absolute,
+            value,
+        )
+    }
+
+    #[inline]
```

Comment:
From what I read about it, adding the `#[inline]` annotation is only necessary/advisable for functions that are used across crates.

Reply:
True, I'll drop them.